R/credentials_gce.R

```diff
@@ -74,36 +74,60 @@
 #' }
 credentials_gce <- function(scopes = "https://www.googleapis.com/auth/cloud-platform",
                             service_account = "default", ...) {
-  gargle_debug("trying {.fun credentials_gce}")
+  gargle_debug("Trying {.fun credentials_gce} ...")
   if (!is_gce()) {
+    gargle_debug(c("x" = "We don't seem to be on GCE."))
     return(NULL)
   }
 
-  gce_token <- fetch_gce_access_token(scopes, service_account = service_account)
+  requested_scopes <- normalize_scopes(scopes)
+  dat <- gce_instance_service_accounts()
+  service_account_details <- as.list(dat[dat$name == service_account, ])
+
+  account_scopes <- service_account_details$scopes
+  account_scopes <- normalize_scopes(strsplit(account_scopes, split = ",")[[1]])
+  missing <- setdiff(requested_scopes, account_scopes)
+  if (length(missing) > 0) {
+    gargle_debug(c(
+      "!" = "{cli::qty(length(missing))}{?This/These} requested \\
+             scope{?s} {?is/are} not among the scopes for the \\
+             {.val {service_account}} service account:",
+      bulletize(missing, bullet = "x"),
+      "i" = "If there are problems downstream, this might be the root cause."
+    ))
+  }
```

Comment:
This whole scope checking block is new.

Pragmatically, I could do this comparison (instance scopes vs. requested scopes) only when the `"gargle_verbosity"` option is `"debug"`. But it feels like a better practice to use the option in the narrowest sense, i.e. only to control messaging.